From d15aca8db281db2a8942629b13fc0cbb19b8eb3b Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Mon, 11 Oct 2021 09:27:25 +0200
Subject: [PATCH 142/529] ARM: dts: stm32: add Hardware debug port (HDP)
 pinctrl to stm32mp15

Add pinctrl definition for The Hardware Debug Port which allows
the observation of internal signals.
By using multiplexers, up to 16 signals for each of 8-bit output
can be observed.

Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: I5ebabfcd2fa4ac30694c7a272ef138690ffc2ce3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/279364
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Antonio Maria BORNEO <antonio.borneo@st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm/boot/dts/stm32mp15-pinctrl.dtsi | 45 ++++++++++++++++++++++++
 1 file changed, 45 insertions(+)

--- a/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
+++ b/arch/arm/boot/dts/stm32mp15-pinctrl.dtsi
@@ -549,6 +549,51 @@
 		};
 	};
 
+	hdp0_pins_a: hdp0-0 {
+		pins {
+			pinmux = <STM32_PINMUX('I', 12, AF2)>; /* HDP0 */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <2>;
+		};
+	};
+
+	hdp0_sleep_pins_a: hdp0-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('I', 12, ANALOG)>; /* HDP0 */
+		};
+	};
+
+	hdp6_pins_a: hdp6-0 {
+		pins {
+			pinmux = <STM32_PINMUX('K', 5, AF2)>; /* HDP6 */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <2>;
+		};
+	};
+
+	hdp6_sleep_pins_a: hdp6-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('K', 5, ANALOG)>; /* HDP6 */
+		};
+	};
+
+	hdp7_pins_a: hdp7-0 {
+		pins {
+			pinmux = <STM32_PINMUX('K', 6, AF2)>; /* HDP7 */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <2>;
+		};
+	};
+
+	hdp7_sleep_pins_a: hdp7-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('K', 6, ANALOG)>; /* HDP7 */
+		};
+	};
+
 	i2c1_pins_a: i2c1-0 {
 		pins {
 			pinmux = <STM32_PINMUX('D', 12, AF5)>, /* I2C1_SCL */
